sync2/p2p_test.go

```diff
@@ -156,3 +157,55 @@ func TestP2P(t *testing.T) {
 			}))
 	}
 }
+
+func TestConfigValidation(t *testing.T) {
+	for _, tc := range []struct {
+		name   string
+		cfg    func(cfg *sync2.Config)
+		expErr string
+	}{
+		{
+			name: "default",
+			cfg:  func(cfg *sync2.Config) {},
+		},
+		{
+			name: "empty",
+			cfg: func(cfg *sync2.Config) {
+				*cfg = sync2.Config{}
+			},
+			expErr: "max-send-range must be positive",
+		},
+		{
+			name: "bad toplevel config",
+			cfg: func(cfg *sync2.Config) {
+				cfg.SyncInterval = 0
+			},
+			expErr: "sync-interval must be positive",
+		},
+		{
+			name: "bad range set reconciler config",
+			cfg: func(cfg *sync2.Config) {
+				cfg.RangeSetReconcilerConfig.MaxSendRange = 0
+			},
+			expErr: "max-send-range must be positive",
+		},
+		{
+			name: "bad multi peer reconciler config",
+			cfg: func(cfg *sync2.Config) {
+				cfg.MultiPeerReconcilerConfig.MaxSyncDiff = -1
+			},
+			expErr: "max-sync-diff must be non-negative",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := sync2.DefaultConfig()
+			tc.cfg(&cfg)
+			err := cfg.Validate()
+			if tc.expErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tc.expErr)
+			}
```

Comment:
NIT: Avoid branches in test cases these make changing the code that is covered by them more complicated - instead I suggest to have a dedicated `t.Run` for the `"default"` test case.

Reply:
Fixed